cobbler.spec

```diff
@@ -384,6 +384,40 @@ chgrp %{apache_group} %{_sysconfdir}/cobbler/settings.yaml
 %else
 %post
 %systemd_post cobblerd.service
+
+if [ $1 -gt 1 ]; then
+    if [ -f /etc/susemanager-release ] || [ -f /etc/uyuni-release ]; then
+        if [ -f /etc/cobbler/settings.yaml.rpmnew ] && [ ! -f /etc/cobbler/settings.d/zz-uyuni.settings ]; then
+            # The /etc/cobbler/settings.yaml.backup should be existing unless explicitely removed.
+            if [ ! -f /etc/cobbler/settings.yaml.backup ]; then
+                echo "Not found: /etc/cobbler/settings.yaml.backup - Skipping settings migration"
+            else
+                python3 <<EOF
+# Check changes between old defaults (backup) and current settings
+# to detect local changes applied to settings. Local changes are then
+# moved to /etc/cobbler/settings.d/zz-uyuni.settings
+import yaml
+old_backup_settings = None
+current_settings = None
+new_uyuni_settings = {}
+with open("/etc/cobbler/settings.yaml.backup") as _input:
+    old_backup_settings = yaml.safe_load(_input)
+with open("/etc/cobbler/settings.yaml") as _input:
+    current_settings = yaml.safe_load(_input)
+for conf in current_settings:
+    if current_settings[conf] != old_backup_settings[conf]:
+       new_uyuni_settings[conf] = current_settings[conf]
+with open("/etc/cobbler/settings.d/zz-uyuni.settings", "w") as _output:
+    yaml.safe_dump(new_uyuni_settings, _output)
+EOF
+                # Restore the default Cobbler settings.yaml file
+                cp /etc/cobbler/settings.yaml /etc/cobbler/settings.yaml.migration.backup
```

Comment:
Should we now declare the `settings.yaml.migration.backup` file in the `%files` section, so that uninstalling Cobbler removes the `.backup` file as well? 

Not sure if that's handled automatically in the spec file already.